client/src/routes/HomePage.tsx

```diff
@@ -1,68 +1,270 @@
-                Welcome, {username}!
-            </h1>
-            <HomePageLinks RouteInfo={tempRouteInfo}/>
-        </div>
-    )
+  const createUser = async () => {
+
+    // duplicate email toast condition
+    const userEmail = JSON.stringify({
+        user_email: email,
+    });
+
+    const userEmailResponse = await fetch("/api/userByEmail", {
+      method: "post",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+      },
+      body: userEmail,
+    });
+    const userEmailStatus = await userEmailResponse.json();
+
+    if (userEmailStatus.length > 0) {
+      toast.error(email + " is already in use.");
+    }
+
+    // invalid email toast condition
+    if (!email.includes("@")) {
+      toast.warning("Email format is invalid.");
+      return;
+    }
+
+    // matching passwords toast condition
+    if (password !== verifyPassword) {
+      toast.warning("Please make sure your passwords match.");
+      return;
+    }
 
-    // return (
-    //     <div className="h-screen flex items-center justify-center">
-    //         <table>
-    //             <tr className="text-3xl">
-    //                 <th>Name</th>
-    //                 <th>Location</th>
-    //                 <th>Website</th>
-    //             </tr>
-    //             {collegeTable.map((data) => (
-    //                 <tr>
-    //                     <td className="w-48 text-center text-2xl p-4">{data.college_name}</td>
-    //                     <td className="w-48 text-center text-2xl p-4">{data.location_city}, {data.location_state}</td>
-    //                     <td className="w-48 text-center text-2xl p-4">{data.general_web_addr}</td>
-    //                 </tr>
-    //             ))
-    //             }
-    //         </table>
-    //     </div>
-    // );
+    const userBody = JSON.stringify({
+      userData: {
+        user_email: email,
+        user_password: password,
+        user_firstname: firstName,
+        user_lastname: lastName,
+      },
+    });
+
+    const response = await fetch("/api/createUser", {
+      method: "post",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+      },
+      body: userBody,
+    });
+    const status = await response.json();
+    console.log(status);
+    //would put toast here
+    setTabs("Logged in and user created");
+  };
+
+  const authUser = async () => {
+    // invalid email toast condition
+    if (!email.includes("@")) {
+      toast.warning("Email format is invalid.");
+      return;
+    }
+
+    const authBody =  JSON.stringify({
+      email: email,
+      password: password,
+    })
+    const response = await fetch("/api/validateUser", {
+      method: "post",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+      },
+      body: authBody
+    });
+    const status = await response.json();
+    if (status <= 0) {
```

Comment:
The status is returned in the format [user_id, user_status] so you may need to use something like status[1] <= 0 instead